File: src/client/point.cpp
#include <iostream>
#include <sstream>
#include <cstring>
#include <vector>
#include <tgmath.h>
#include "point.h"

using namespace std;

Point::Point(int x, int y) : x_(x), y_(y) {}

Point::~Point() {}

int Point::getX() const {
  return x_;
}

string Point::toString() {
  ostringstream ss;
  ss << x_ << " " << y_;
  string s = ss.str();
  return s;
}

int Point::getY() const {
  return y_;
}

void Point::setX(int x) {
  x_ = x;
}

void Point::setY(int y) {
  y_ = y;
}

Point Point::decrease() {
  return Point(x_-1, y_-1);
}

bool Point::operator==(const Point &point) const {
  return x_ == point.getX() && y_ == point.getY();
}

bool Point::operator!=(const Point &point) const {
  return x_ != point.getX() || y_ != point.getY();
}

Point::Point(char point[]) {
  x_ = 0;
  y_ = 0;
  int i = 0;
  while (point[i] != ' ') {
    x_ *= 10;
    x_ += (point[i] - 48);
    i++;
  }
  i++;
  while (point[i] != '\0') {
    y_ *= 10;
    y_ += (point[i] - 48);
    i++;
  }
}

std::ostream &operator<<(std::ostream &out, Point const &point) {
  return out << "(" << point.getX() + 1 << ", " << point.getY() + 1 << ")";
}

File: Console/Point.java
package Console;

public class Point {
    private int x;
    private int y;

    /**
     * Constructor for point by coordinates.
     *
     * @param x coordinate for point.
     * @param y coordinate for point.
     */
    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    /**
     * Sets x coordinate of the point.
     *
     * @return x coordinate of the point.
     */
    public int getX() {
        return this.x;
    }

    /**
     * Sets y coordinate of the point.
     *
     * @param x y coordinate of the point.
     */
    public void setX(int x) {
        this.x = x;
    }

    /**
     * Returns y coordinate of the point.
     *
     * @return y coordinate of the point.
     */
    public int getY() {
        return this.y;
    }

    /**
     * Returns x coordinate of the point.
     *
     * @return x coordinate of the point.
     */
    public void setY(int y) {
        this.y = y;
    }

    /**
     * Makes the string printable.
     *
     * @return String the printable string.
     */
    public String toString() {
        return "(" + this.x + ", " + this.y + ")";
    }
}